inmoose/pycombat/covariates.py

```diff
@@ -83,15 +86,23 @@ def __init__(self, counts, batch, covar_mod=None, ref_batch=None):
             list_samples = counts.columns
             list_genes = counts.index
             counts = counts.values
-            dataframe_instance = True
+            input_type = "dataframe"
         elif isinstance(counts, np.ndarray):
             list_samples = None
             list_genes = None
-            dataframe_instance = False
+            input_type = "ndarray"
+        elif isinstance(counts, AnnData):
+            list_samples = counts.obs_names
+            list_genes = counts.var_names
+            self.input_ad = counts
+            counts = counts.X.T
+            input_type = "anndata"
+            if isinstance(batch, str):
+                batch = self.input_ad.obs[batch]
```

Comment:
What if the user forgot to add the batch in the obs? Maybe a try except is needed